pywr-schema/src/nodes/core.rs

```diff
@@ -709,11 +745,37 @@ impl AggregatedNode {
 
 #[cfg(feature = "core")]
 impl AggregatedNode {
-    pub fn add_to_model(&self, network: &mut pywr_core::network::Network) -> Result<(), SchemaError> {
-        let nodes = self
+    pub fn node_indices_for_constraints(
+        &self,
+        network: &pywr_core::network::Network,
+        args: &LoadArgs,
+    ) -> Result<Vec<pywr_core::node::NodeIndex>, SchemaError> {
+        let indices = self
             .nodes
             .iter()
-            .map(|name| network.get_node_index_by_name(name, None))
+            .map(|name| {
+                args.schema
+                    .get_node_by_name(name)
+                    .ok_or_else(|| SchemaError::NodeNotFound(name.to_string()))?
+                    .node_indices_for_constraints(network, args)
+            })
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .flatten()
+            .collect();
+        Ok(indices)
+    }
+    pub fn add_to_model(&self, network: &mut pywr_core::network::Network, args: &LoadArgs) -> Result<(), SchemaError> {
+        let nodes: Vec<Vec<_>> = self
```

Comment:
No, it's slightly different. This version is a nested vec. Each entry of the outer vec is a vec of that node's indices (i.e. it could refer to multiple core nodes). The above function is a flattened version of this array.